packages/e2e/test/ws-server/webSocket.test.ts

```diff
@@ -1,5 +1,195 @@
-describe('ws-server', () => {
-  it('tmp', () => {
-    expect(1).toBeTruthy();
+import { CardanoWsClient } from '@cardano-sdk/cardano-services-client';
+import {
+  CardanoWsServer,
+  GenesisData,
+  createDnsResolver,
+  getOgmiosCardanoNode,
+  util
+} from '@cardano-sdk/cardano-services';
+import { HealthCheckResponse, WsProvider } from '@cardano-sdk/core';
+import { OgmiosCardanoNode } from '@cardano-sdk/ogmios';
+import { Pool } from 'pg';
+import { getEnv, walletVariables } from '../../src';
+import { getPort } from 'get-port-please';
+import { logger } from '@cardano-sdk/util-dev';
+
+const env = getEnv([...walletVariables, 'DB_SYNC_CONNECTION_STRING', 'OGMIOS_URL']);
+
+const wsProviderReady = (provider: WsProvider) =>
+  new Promise<void>((resolve, reject) => {
+    // eslint-disable-next-line prefer-const
+    let timeout: NodeJS.Timeout | undefined;
+
+    const subscription = provider.health$.subscribe(({ ok }) => {
+      if (ok) {
+        if (timeout) clearTimeout(timeout);
+        subscription.unsubscribe();
+        resolve();
+      }
+    });
+
+    timeout = setTimeout(() => {
+      subscription.unsubscribe();
+      reject(new Error('WsProvider timeout'));
+    }, 10_000);
+
+    timeout.unref();
+  });
+
+const wsProviderReadyAgain = (provider: WsProvider, close: () => Promise<unknown>) =>
+  new Promise<void>(async (resolve, reject) => {
+    const oks: boolean[] = [];
+    let closed = false;
+    // eslint-disable-next-line prefer-const
+    let timeout: NodeJS.Timeout | undefined;
+
+    const subscription = provider.health$.subscribe(({ ok }) => {
+      oks.push(ok);
+
+      if (closed && ok) {
+        if (timeout) clearTimeout(timeout);
+        subscription.unsubscribe();
+
+        try {
+          const [, ...last] = oks;
+
+          // The first emitted event is the ok buffered one
+          // next we expect at least one not ok event i.e. the close function had the desired effect
+          // last we expect one more ok event when provider is operational once again
+          expect(oks.length).toBeGreaterThanOrEqual(3);
+          expect(oks[0]).toBeTruthy();
+          expect(oks[1]).toBeFalsy();
+          expect(last.some((result) => result));
+
+          resolve();
+        } catch (error) {
+          reject(error);
+        }
+      }
+    });
+
+    try {
+      await close();
+    } catch (error) {
+      reject(error);
+    }
+
+    closed = true;
+
+    timeout = setTimeout(() => {
+      subscription.unsubscribe();
+      reject(new Error('WsProvider timeout'));
+    }, 10_000);
+
+    timeout.unref();
+  });
+
+describe('Web Socket', () => {
```

Comment:
I'd like to see a test for some network info provider failure due to connection issue, because that is quite a lot of untested code/logic

Reply:
d39330353d and 7a3447e3a9 should solve this